napari/layers/_vectors_layer/model.py

```diff
@@ -0,0 +1,461 @@
+        Input coordinate of (N,4) becomes two output coordinates of (N,2)
+
+        :param vect: np.ndarray of shape (N, 4)
+        :return: position list of shape (2*N, 2) for vispy
+        """
+
+        # create empty vector of necessary shape
+        #   one coordinate for each endpoint of the vector
+        pos = np.empty((2 * len(vect), 2), dtype=np.float32)
+
+        # create pairs of points
+        pos[0::2, 0] = vect[:, 0]
+        pos[1::2, 0] = vect[:, 0]
+        pos[0::2, 1] = vect[:, 1]
+        pos[1::2, 1] = vect[:, 1]
+
+        # adjust second of each pair according to x-y projection
+        pos[1::2, 0] += vect[:, 2]
+        pos[1::2, 1] += vect[:, 3]
+
+        return pos
+
+    @property
+    def averaging(self) -> str:
+        """
+        Set the kernel over which to average
+        :return: string of averaging kernel size
+        """
+        return self._averaging
+    
+    @averaging.setter
+    def averaging(self, value: int):
+        """
+        Calculates an average vector over a kernel
+        :param averaging: one of "_avg_dims" above
+        :return:
+        """
+        self._averaging = value
+        self._kernel = value
+
+        # emit signal back to qt_properties for averaging
+        self.events.emit_avg()
+
+        self._refresh()
+
+    def averaging_bind_to(self, callback):
+        """
+        register an observer to be notified upon changes to averaging
+        Removes the default method for averaging if an external method
+            is registered
+        :param callback: function to call upon averaging changes
+        :return:
+        """
+
+        # if self._qt_properties._default_avg in self.events.callbacks:
+        try:
+            print("disconnecting default avg callback")
+            self.events.emit_avg.disconnect(self._qt_properties._default_avg)
+        except Exception as ex:
+            print('error disconnecting avg callback: '+str(ex))
+            pass
+        self.events.emit_avg.connect(callback)
+
+    @property
+    def width(self) -> Union[int, float]:
+        """
+        width of the line in pixels
+            widths greater than 1px only guaranteed to work with "agg" method
+        :return: int or float line width
+        """
+        return self._width
+
+    @width.setter
+    def width(self, width: Union[int, float]) -> None:
+        self._width = width
+        self._refresh()
+
+    @property
+    def length(self) -> Union[int, float]:
+        return self._length
+
+    @length.setter
+    def length(self, length: Union[int, float]):
+        """
+        Change the length of all lines
+        :param length: length multiplicative factor
+        :return: None
+        """
+        self._length = length
+
+        # emit signal back to qt_properties for length
+        self.events.emit_len()
+
+        self._refresh()
+
+    def length_bind_to(self, callback):
```

Comment:
see my comment about adding the averaging binding - i think the same applies here too